fix(guidelines): update bits and bytes guidance (#5572)
scaleway/docs-content: pages/guidelines.mdx

### Bits and bytes

- Add a space between the bits/bytes abbreviation and the number, e.g. `100 MB`, `1 Gbps`.
- Use the following abbreviations:

| Abbreviation     | Unit                           | Used for                                   |  Example                             |
|------------------|--------------------------------|--------------------------------------------|--------------------------------------|
| MB, GB, TB       | Megabyte / Gigabyte / Terabyte | Storage capacity, RAM, VRAM, disk size     | Storage: 32 GB           |
| MiB, GiB, TiB    | Mebibyte / Gibibyte / Tebibyte | RAM / memory in binary multiples where technical precision is important  | RAM: 512 MiB  |
| Mbps, Gbps       | Megabit per second / Gigabit per second | Network bandwidth (bits per second) | Bandwidth: up to 10 Gbps          |
| MB-s, GB-s       | Megabyte seconds / Gigabyte seconds | Cumulative data transfer/consumption over time | Memory consumption: €0.10 per 100 000 GB-s |

<Message type="note">
GB vs GiB is an SI vs binary distinction:
- GB = 10⁹ bytes = 1,000,000,000 bytes 
- GiB = 2³⁰ bytes = 1,073,741,824 bytes
<br/><br/>

> Use GiB/MiB for memory in contexts where technical precision and binary vs decimal distinction is critical. In other contexts, GB/MB may be appropriate for memory.

Author: ```suggestion
Use GiB/MiB in contexts where technical precision and binary vs decimal distinction is critical, particularly for memory.
```
